libs/cardano-ledger-test/src/Test/Cardano/Ledger/Generic/Consensus.hs

```diff
@@ -0,0 +1,1166 @@
+      sgSlotLength = secondsToNominalDiffTime 2,
+      sgUpdateQuorum = quorum testGlobals,
+      sgMaxLovelaceSupply = maxLovelaceSupply testGlobals,
+      sgProtocolParams = emptyPParams,
+      sgGenDelegs = Map.empty,
+      sgInitialFunds = mempty,
+      sgStaking = emptyGenesisStaking
+    }
+
+testEpochInfo :: EpochInfo Identity
+testEpochInfo = epochInfoPure testGlobals
+
+-- ================================================================================
+
+-- | Build a Tx from a list of maybe fields. The maybe is usfull because sometimes we
+--   only want to add a field in a particular era, so we might say something like this
+-- genericTx proof
+--  [Just field1
+--  , if (Some proof) >= Mary Mock then Just field2 else Nothing
+--  , Just field3
+--  ]
+genericTx :: Proof era -> [Maybe (TxField era)] -> Core.Tx era
+genericTx proof xs = newTx proof (foldl' accum [] xs)
+  where
+    accum ans Nothing = ans
+    accum ans (Just field) = field : ans
+
+-- | Build a Tx from a list of maybe fields. The maube is usfull because sometimes we
+--   only want to add a field in a particular era, so we might say something like this
+-- genericTxBody proof
+--  [Just field1
+--  , if (Some proof) >= Mary Mock then Just field2 else Nothing
+--  , Just field3
+--  ]
+genericTxBody :: Era era => Proof era -> [Maybe (TxBodyField era)] -> Core.TxBody era
+genericTxBody proof xs = specialize @Core.EraTxBody proof (newTxBody proof (foldl' accum [] xs))
+  where
+    accum ans Nothing = ans
+    accum ans (Just field) = field : ans
+
+genericTxOut :: Era era => Proof era -> [Maybe (TxOutField era)] -> Core.TxOut era
+genericTxOut proof xs = newTxOut proof (foldl' accum [] xs)
+  where
+    accum ans Nothing = ans
+    accum ans (Just field) = field : ans
+
+genericWits :: Era era => Proof era -> [Maybe (WitnessesField era)] -> Core.TxWits era
+genericWits proof xs = newWitnesses merge proof (foldl' accum [] xs)
+  where
+    accum ans Nothing = ans
+    accum ans (Just field) = field : ans
+
+-- =========================================================
+-- Individual examples for each Era
+-- =========================================================
+
+type StandardShelley = ShelleyEra StandardCrypto
+
+-- | ShelleyLedgerExamples for Shelley era
+ledgerExamplesShelley :: ShelleyLedgerExamples StandardShelley
+ledgerExamplesShelley =
+  defaultLedgerExamples
+    (Shelley Standard)
+    exampleCoin
+    (exampleTxBody (Shelley Standard) (inject (Coin 100000)))
+    exampleAuxiliaryDataShelley
+    ()
+
+exampleCoin :: Coin
+exampleCoin = Coin 10
+
+exampleMetadataMap :: Map Word64 Metadatum
+exampleMetadataMap =
+  Map.fromList
+    [ (1, S "string"),
+      (2, B "bytes"),
+      (3, List [I 1, I 2]),
+      (4, Map [(I 3, B "b")])
+    ]
+
+exampleAuxiliaryDataShelley :: Core.AuxiliaryData StandardShelley
+exampleAuxiliaryDataShelley = Metadata exampleMetadataMap
+
+-- ======================
+
+type StandardAllegra = AllegraEra StandardCrypto
```

Comment:
```suggestion
```

Reply:
This mimics the code we are trying to replace. For each Era they make a type synonym. This is done in this file for all the other Eras. So I will leave it for consistency.